src/SecureStore.HashicorpVault/HashicorpVaultSecureStore.cs

```diff
@@ -0,0 +1,381 @@
+                    DisplayName = HashicorpVaultUtils.GetLocalizedResource(nameof(Resource.SettingVaultUri)),
+                    IsMandatory = true,
+                },
+                new ConfigurationValue(ConfigurationValueType.Choice)
+                {
+                    Key = "AuthenticationType",
+                    DisplayName = HashicorpVaultUtils.GetLocalizedResource(nameof(Resource.SettingAuthenticationType)),
+                    IsMandatory = true,
+                    PossibleValues = Enum.GetNames(typeof(AuthenticationType)),
+                },
+                new ConfigurationSection
+                {
+                    Key = nameof(AuthenticationType.AppRole),
+                    DisplayName = nameof(AuthenticationType.AppRole),
+                    Configurations = new[]
+                    {
+                        new ConfigurationValue(ConfigurationValueType.Secret)
+                        {
+                            Key = "RoleId",
+                            DisplayName = HashicorpVaultUtils.GetLocalizedResource(nameof(Resource.SettingRoleId)),
+                            IsMandatory = false,
+                        },
+                        new ConfigurationValue(ConfigurationValueType.Secret)
+                        {
+                            Key = "SecretId",
+                            DisplayName = HashicorpVaultUtils.GetLocalizedResource(nameof(Resource.SettingSecretId)),
+                            IsMandatory = false,
+                        },
+                    },
+                },
+                new ConfigurationSection
+                {
+                    Key = $"{nameof(AuthenticationType.UsernamePassword)}/{nameof(AuthenticationType.Ldap)}",
+                    DisplayName = nameof(AuthenticationType.ClientCertificate),
+                    Configurations = new[]
+                    {
+                        new ConfigurationValue(ConfigurationValueType.String)
+                        {
+                            Key = "Username",
+                            DisplayName = HashicorpVaultUtils.GetLocalizedResource(nameof(Resource.SettingUsername)),
+                            IsMandatory = false,
+                        },
+                        new ConfigurationValue(ConfigurationValueType.Secret)
+                        {
+                            Key = "Password",
+                            DisplayName = HashicorpVaultUtils.GetLocalizedResource(nameof(Resource.SettingPassword)),
+                            IsMandatory = false,
+                        },
+                    },
+                },
+                new ConfigurationSection
+                {
+                    Key = nameof(AuthenticationType.ClientCertificate),
+                    DisplayName = nameof(AuthenticationType.ClientCertificate),
+                    Configurations = new[]
+                    {
+                        new ConfigurationValue(ConfigurationValueType.Secret)
```

Comment:
This fits better as a `SecretFile`, but support for it did not yet land in 20.10, it will probably be added later. But adding proper support for this auth type is complicated (see the CyberArkCCP plugin). If not needed by customers right now, I would prefer to not add support for this at all. We should recommend using `AppRole` anyway.

Reply:
Removed auth type